### PR TITLE
[DOI-842] Use Link component when redirecting inside Webapp for breadcrumb

### PR DESCRIPTION
**Pull request by fgchaio**
These changes are related to [this discussion](https://github.com/FromDoppler/doppler-webapp/pull/2221#discussion_r1062365354) we had.

Looking forward to your review!

### CODE REVIEW

File: src/components/shared/Breadcrumb/Breadcrumb.js [hunk 1]
```diff
@@ -12,9 +13,17 @@ const BreadcrumbItem = ({ href, text }) => {
   return (
     <>
       {!!href ? (
-        <li>
-          <a href={href}>{text}</a>
-        </li>
+        href.startsWith('http') ? (
```
Comment (andresmoschini) [inline src/components/shared/Breadcrumb/Breadcrumb.js:16]: I think that it follows the meaning more accurately:

```suggestion
        /^(?:[a-z]+:)?\/\//.test(href) ? (
```

Absolute: 
* `http://www.google.com`
* `ftp://www.google.com/folder`
* `//www.google.com`

Relative:
* `/folder`
* `folder`
* `folder.name`

File: src/components/shared/Breadcrumb/Breadcrumb.js [hunk 2]
```diff
@@ -12,9 +13,17 @@ const BreadcrumbItem = ({ href, text }) => {
...
+          <li>
+            <a href={href}>{text}</a>
+          </li>
+        ) : (
+          <li>
+            <Link to={href} data-testid="internal-link">
+              {text}
+            </Link>
+          </li>
```
Comment (andresmoschini) [inline src/components/shared/Breadcrumb/Breadcrumb.js:25]: What do you think about moving the `<li>` out of the condition?

```suggestion
        <li>
          href.startsWith('http') ? (
            <a href={href}>{text}</a>
          ) : (
            <Link to={href} data-testid="internal-link">
              {text}
            </Link>
          )          
        </li>
```

In a next step, we could replace it with a smarter component:

```suggestion
        <li>
          <SmartLink to={href}>
            {text}
          </SmartLink>
        </li>
```

Comment (fgchaio) [reply src/components/shared/Breadcrumb/Breadcrumb.js:25]: good idea!

**Comment by andresmoschini** [comment]
🚀

**Pull request merged**